Formula/beagle.rb

```diff
@@ -22,19 +22,20 @@ class Beagle < Formula
     sha256 cellar: :any_skip_relocation, x86_64_linux:   "1af7280eaec10e6a5e335326793ca36e36e8b41e507ea891ff3bbfdb3d453d01"
   end
 
-  depends_on "autoconf" => :build
-  depends_on "automake" => :build
-  depends_on "doxygen" => :build
-  depends_on "libtool" => :build
+  depends_on "cmake" => :build
   depends_on "openjdk" => [:build, :test]
 
   def install
-    args = std_configure_args + %w[--without-cuda --disable-libtool-dev]
-    args << "--disable-sse" if Hardware::CPU.arm?
-
-    system "./autogen.sh"
-    system "./configure", *args
-    system "make", "install"
+    mkdir "build" do
+      system "cmake",
+        "-DJAVA_HOME=#{Formula["openjdk"].opt_prefix}",
+        "-DJAVA_AWT_LIBRARY=#{Formula["openjdk"].include}",
+        "-DJAVA_JVM_LIBRARY=#{Formula["openjdk"].include}",
+        "-DCMAKE_INSTALL_PREFIX:PATH=#{prefix}",
+        ".."
+      system "make"
+      system "make", "install"
+    end
```

Comment:
Let's follow the pattern used by `brew create --cmake`.
```suggestion
    system "cmake", "-S", ".", "-B", "build",
                    "-DJAVA_HOME=#{Formula["openjdk"].opt_prefix}",
                    "-DJAVA_AWT_LIBRARY=#{Formula["openjdk"].include}",
                    "-DJAVA_JVM_LIBRARY=#{Formula["openjdk"].include}",
                    *std_cmake_args
    system "cmake", "--build", "build"
    system "cmake", "--install", "build"
```